### ISSUE TITLE
WPT test failure expectations shouldn't use an index in the test name

### ISSUE THREAD

**Issue by brainlessdeveloper** [servo/servo#20668]
I just had the following situation in this PR: https://github.com/servo/servo/pull/20631

My PR adds a WebGL method `getRenderbufferParameter`, and this made a test pass. The test names for WebGL have indeces in their names. My new "unexpected" test failures looked like this:

```
  FAIL WebGL test #0: Property either does not exist or is not a function: getUniform      
  FAIL WebGL test #1: Property either does not exist or is not a function: isContextLost                                                                                                                         
  FAIL WebGL test #2: Property either does not exist or is not a function: getBufferSubData   
  FAIL WebGL test #3: Property either does not exist or is not a function: copyBufferSubData  
  FAIL WebGL test #4: Property either does not exist or is not a function: blitFramebuffer      
  FAIL WebGL test #5: Property either does not exist or is not a function: framebufferTextureLayer        
  FAIL WebGL test #6: Property either does not exist or is not a function: getInternalformatParameter
```

Actually, there were 90 of those, but for brevity...

Currently, the failure expectations in the `.ini` file look like this:
```
  [WebGL test #1: Property either does not exist or is not a function: getUniform]
    expected: FAIL

  [WebGL test #2: Property either does not exist or is not a function: isContextLost]
    expected: FAIL

  [WebGL test #3: Property either does not exist or is not a function: getBufferSubData]
    expected: FAIL

  [WebGL test #4: Property either does not exist or is not a function: copyBufferSubData]
    expected: FAIL

  [WebGL test #5: Property either does not exist or is not a function: blitFramebuffer]
    expected: FAIL

  [WebGL test #6: Property either does not exist or is not a function: framebufferTextureLayer]
    expected: FAIL
```

They were also 90. If you notice the difference, the failure expectations start with a `#1` and the new failures start with a `#0`. This is because of my PR, which fixes a test that presumably comes before the first one in this list.

This is really bad and indeces should not be used in test names, to avoid confusion for future developers.

**Comment by nox** [comment]
Yeah, it's a problem with the tests directly in upstream, we can't just modify them here. I usually delete the expectation entirely and then I update it.

**Comment by gootorov** [comment]
There's a nice script that does that:
[updating test expectations](https://github.com/servo/servo/tree/master/tests/wpt#updating-test-expectations)

**Comment by brainlessdeveloper** [comment]
Thanks! That kind of solves it.

**Comment by asajeffrey** [comment]
This issue is so bad that on some tests, AFAICT no matter what you put in the .ini file, the test doesn't meet expectations.

I ended up disabling `/_webgl/conformance/textures/misc/copy-tex-image-and-sub-image-2d.html` on macos because of this.

**Comment by jdm** [comment]
The index is something that we added as part of our automated importing of the khronos test harness to WPT-ify it: https://github.com/servo/servo/blob/master/tests/wpt/webgl/tools/js-test-pre.patch#L11

It's necessary (as far as I recall) because the test messages that we use from the khronos tests are not always unique. One option might be to store a hash of all previously-encountered messages and only add an index when there would otherwise be a duplicate message, so that (hopefully) most messages wouldn't include it.

**Comment by jdm** [comment]
I have a suspicion that the problems with copy-tex-image-and-sub-image-2d.html have less to with the indexes and more to do with the test names including x/y values - if the test ends up reading uninitialized texture memory, the results would be nondeterministic, and the presence of an index wouldn't make that any worse.

**Comment by asajeffrey** [comment]
Oh that's true, the test name includes other data generated at runtime from the WebGL state, it's not just the index.

**Comment by asajeffrey** [comment]
I created a more general issue for nondeterministic test names.
